perspective_reddit_bot/moderate_subreddit.py

```diff
@@ -199,49 +245,40 @@ def score_subreddit(creds_dict,
   perspective = perspective_client.PerspectiveClient(
     creds_dict['perspective_api_key'])
 
+  current_file_time = datetime.utcnow().strftime('%Y%m%d_%H%M%S')
   for i, comment in enumerate(subreddit.stream.comments()):
     try:
-      # TODO(nthain): Make this loop more readable by breaking into functions.
       if i % 100 == 0 and i > 0:
         print(i)
         # Check if still has mod permissions every 100 comments
         mod_permissions = bot_is_mod(reddit, subreddit)
 
-      original_comment = comment.body
-      comment_for_scoring = (remove_quotes(original_comment)
-                             if should_remove_quotes else original_comment)
-      if len(comment_for_scoring) > 20000:
-        print('Comment too long, skipping...')
+      # Score current comment with perspective models
+      comment_for_scoring, scores = score_comment(comment,
+                                                  perspective,
+                                                  api_models,
+                                                  ensembles,
+                                                  should_remove_quotes)
+      if scores is None:
         continue
-      scores = perspective.score_text(comment_for_scoring, api_models,
-                                      language=LANGUAGE)
-      for e in ensembles:
-        scores[e.name] = e.predict_one(scores)
 
-      action_dict = defaultdict(list)
-      for rule in rules:
-        if rule.check_model_rules(scores):
-          action_dict[rule.action_name].append(rule.rule_description)
-          print_moderation_decision(i, comment, rule)
+      # Check which rules should be applied
+      action_dict = check_rules(i, comment, rules, scores)
 
+      # Apply actions from triggered rules
       if mod_permissions and original_mod_permissions:
         for action, rule_strings in action_dict.items():
           apply_action(action, comment, rule_strings)
 
-      if output_path:
-        with open(output_path, 'a') as f:
-          record = {
-              'comment_id': comment.id,
-              'comment_text': original_comment,
-              'scored_comment_text': comment_for_scoring,
-              'created_utc': comment.created_utc,
-              'permalink': 'https://reddit.com' + comment.permalink,
-              'author': comment.author.name,
-          }
-          record.update(action_dict)
-          record.update(scores)
-          json.dump(record, f)
-          f.write('\n')
+      # Maybe write comment scores to file
+      if output_dir:
+        file_suffix = '%s_%s.json' % (subreddit_name, current_file_time)
+        output_path = os.path.join(output_dir, file_suffix)
```

Comment:
We're recomputing this every iteration but it doesn't change. Let's lift these 2 lines outside the loop? Like:

```
output_path = None
if output_dir:
  timestamp = timestamp_string(datetime.utcnow())
  output_path = os.path.join(output_dir, '%s_%s.json'.format(subreddit_name, timestamp)
for comment in stream:
  [...]
  if output_path:
    append_comment_data(output_path, ...)
```

Reply:
Done.